src/platform/intel/cavs/lib/pm_runtime.c

```diff
@@ -321,8 +321,20 @@ static inline void cavs_pm_runtime_dis_dsp_pg(uint32_t index)
 		lps_ctl |= SHIM_LPSCTL_FDSPRUN;
 		shim_write(SHIM_LPSCTL, lps_ctl);
 	} else {
+		/* Slave core power up */
 		shim_write16(SHIM_PWRCTL, shim_read16(SHIM_PWRCTL) |
-			     SHIM_PWRCTL_TCPDSPPG(index));
+			     SHIM_PWRCTL_TCPDSPPG(index) |
+			     SHIM_PWRCTL_TCPCTLPG);
+
+		/* Waiting for power up */
+		while (((shim_read16(SHIM_PWRSTS) & SHIM_PWRCTL_TCPDSPPG(index)) !=
+			SHIM_PWRCTL_TCPDSPPG(index)) && tries--) {
+			idelay(delay_count);
+		}
+		/* Timeout check with warning log */
+		if (tries == 0)
+			trace_power("cavs_pm_runtime_dis_dsp_pg(): failed to power up core %d",
```

Comment:
trace_power_error() since this is a failure.

Reply:
added trace_power_error()